PowerSwitcher.TrayApp/App.xaml.cs

```diff
@@ -66,24 +67,47 @@ private void Configuration_PropertyChanged(object sender, System.ComponentModel.
                 if (Configuration.Data.ShowOnShortcutSwitch) { registerHotkeyFromConfiguration(); }
                 else { unregisterHotkeyFromConfiguration(); }
             }
+
+            if(e.PropertyName == nameof(PowerSwitcherSettings.CycleNextSchemaSwitch))
+            {
+                if (Configuration.Data.CycleNextSchemaSwitch) { registerHotkeyFromConfiguration(true); }
+                else { unregisterHotkeyFromConfiguration(true); }
+            }
+
         }
 
-        private void unregisterHotkeyFromConfiguration()
+        private void unregisterHotkeyFromConfiguration(bool isCycleNextSchemaSwitch = false)
         {
-            HotKeyManager.Unregister(new HotKey(Configuration.Data.ShowOnShortcutKey, Configuration.Data.ShowOnShortcutKeyModifier));
+            if(isCycleNextSchemaSwitch)
+                HotKeyManager.Unregister(new HotKey(Configuration.Data.CycleNextSchemaKey, Configuration.Data.CycleNextSchemaKeyModifier));
+            else
+                HotKeyManager.Unregister(new HotKey(Configuration.Data.ShowOnShortcutKey, Configuration.Data.ShowOnShortcutKeyModifier));
         }
 
-        private bool registerHotkeyFromConfiguration()
+        private bool registerHotkeyFromConfiguration(bool isCycleNextSchemaSwitch = false)
         {
-            var newHotKey = new HotKey(Configuration.Data.ShowOnShortcutKey, Configuration.Data.ShowOnShortcutKeyModifier);
+            HotKey newHotKey = null;
+            if(isCycleNextSchemaSwitch)
+                newHotKey = new HotKey(Configuration.Data.CycleNextSchemaKey, Configuration.Data.CycleNextSchemaKeyModifier);
+            else
+                newHotKey = new HotKey(Configuration.Data.ShowOnShortcutKey, Configuration.Data.ShowOnShortcutKeyModifier);
 
             bool success = HotKeyManager.Register(newHotKey);
             if(!success) { HotKeyFailed = true; return false; }
-            newHotKey.HotKeyFired += (this.MainWindow as MainWindow).ToggleWindowVisibility;
+
+            if (isCycleNextSchemaSwitch)
+                newHotKey.HotKeyFired += CycleNextPowerSchema;
```

Comment:
The hotkey press is correctly switching the schema to the next power schema.  The problem is that it doesn't show you what schema it switched you to.  What I would like to do is modify `MainWindow.ToggleWindowVisibility()` to sort of act as a toast notification and disappear on its own after, say, 5 seconds.  I can call `newHotKey.HotKeyFired += (this.MainWindow as MainWindow).ToggleWindowVisibility; as it currently is, and it shows the new schema, but it doesn't hide the window unless the user manually clicks somewhere on the screen.  I played around with adding an optional timeout parameter but Visual Studio complained about that.  I also couldn't figure out how to trigger the HideWithAnimation() function after the timeout period.  I'm open to ideas about the best way to implement some sort of indicator notification of what power schema it just got changed to and then letting it disappear... if there is somewhere in the code where you are already doing this, please let me know.

Reply:
Yeah, I vaguely remember the fly-out logic was somehow buggy w.r.t. to timeout parameters. The best way to approach it might be to use winrt API for proper notifications. I.e. instead of creating your own toast-like notification just use native Windows Notification API. 

Not sure how much identity the current packaging schema provides and what's the state of invoking notifications without it, but remember that https://github.com/File-New-Project/EarTrumpet was doing something with winRT APIs so it might be a good place to start. Or you might wait until project reunion comes with notifications API (won't need app identity at all): https://github.com/microsoft/ProjectReunion